font-types/src/name_id.rs

```diff
@@ -153,28 +153,53 @@ impl NameId {
     /// If present in a variable font, it may be used as the family prefix in the PostScript Name Generation
     /// for Variation Fonts algorithm.
     pub const VARIATIONS_POSTSCRIPT_NAME_PREFIX: Self = Self(25);
+
+    /// The last value that is explicitly reserved for standard names.
+    ///
+    /// Values after this are available to be used for font-specific names.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/name#name-ids>
+    pub const LAST_RESERVED_NAME_ID: Self = Self(255);
 }
 
 impl NameId {
     /// Create a new identifier from a raw u16 value.
+    #[inline]
     pub const fn new(raw: u16) -> Self {
         Self(raw)
     }
 
     /// Returns an iterator over the set of predefined identifiers according to the
     /// specification.
+    #[inline]
     pub fn predefined() -> impl Iterator<Item = Self> + Clone {
         // Poor name id 15 got lost...
         (0..15).chain(16..=25).map(Self)
     }
 
     /// Return the identifier as a u16.
+    #[inline]
     pub const fn to_u16(self) -> u16 {
         self.0
     }
 
+    /// Returns `true` if self is in the range `0..=255` (reserved for standard names.)
+    #[inline]
+    pub const fn is_reserved(self) -> bool {
+        self.0 <= Self::LAST_RESERVED_NAME_ID.0
+    }
+
+    /// Returns a new `NameId` by adding `rhs` to the current value.
+    ///
+    /// This is a useful convenience method when constructing a name table.
+    #[inline]
+    pub const fn saturating_add(self, rhs: u16) -> Self {
+        Self(self.0.saturating_add(rhs))
+    }
```

Comment:
Let's just accept it's fallible (however unlikely) and use checked?

Reply:
neat in double-checking this apparently the largest allowed NameID is 32767, so we want to check for that explicitly in any case.